### PR TITLE
6834: Platform JRS apps should run with both JasperDB and MerkleDb data sources

### PR DESCRIPTION
**Pull request by artemananiev**
Fixes: https://github.com/hashgraph/hedera-services/issues/6834
Signed-off-by: Artem Ananev <artem.ananev@swirldslabs.com>

### CODE REVIEW

File: platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/platform/PlatformTestingToolMain.java [hunk 1]
```diff
@@ -711,12 +722,16 @@ private void initBasedOnPayloadCfgSimple(final PayloadCfgSimple pConfig) {
     }
 
     private void initializeAppClient(final String[] pars, final ObjectMapper objectMapper) throws IOException {
```
Comment (artemananiev) [inline platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/platform/PlatformTestingToolMain.java:724]: In `swirlds-platform-regression` repo I couldn't find any test config, which sets more than one param to PTT. It means, this method doesn't seem to be called at all. Still, I decided to leave it as is and use parameter at index 2 for "use MerkleDb" flag

**Review by imalygin — APPROVED** [review]

**Review by timo0 — APPROVED** [review]
Browser LGTM

**Review by poulok — DISMISSED** [review]

File: platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/platform/PlatformTestingToolMain.java [hunk 2]
```diff
@@ -547,12 +547,22 @@ public void init(Platform platform, NodeId id) {
...
                 jsonFileName = parameters[0];
             }
 
+            // Parameters[1]: JSON file for app client (parsed below), optional
+
+            // Parameters[2]: indicates whether to use MerkleDb (true) or JasperDB (false) for virtual maps, optional
+            boolean useMerkleDb = false;
+            if ((parameters != null) && (parameters.length > 2)) {
+                useMerkleDb = Boolean.parseBoolean(parameters[2]);
+            }
```
Comment (timo0) [inline platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/platform/PlatformTestingToolMain.java:563]:
```suggestion
            final boolean useMerkleDb = parameters != null && parameters.length > 2 &&  Boolean.parseBoolean(parameters[2]);
```

**Review by imalygin — APPROVED** [review]

File: platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/platform/PlatformTestingToolMain.java [hunk 3]
```diff
@@ -708,12 +716,16 @@ private void initBasedOnPayloadCfgSimple(final PayloadCfgSimple pConfig) {
...
             return;
         }
 
-        logger.info(LOGM_DEMO_INFO, "Parsing JSON for client ");
         final String jsonFileName = pars[1];
+        if (jsonFileName.trim().isBlank()) {
+            return;
+        }
+
+        logger.info(LOGM_DEMO_INFO, "Parsing JSON for client: {}", jsonFileName);
```
Comment (jsync-swirlds) [inline platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/platform/PlatformTestingToolMain.java:728]: Might be better if this is debug, so we limit console output.

File: platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/virtualmerkle/state/VirtualMerkleStateInitializer.java [hunk 4]
```diff
@@ -67,110 +77,166 @@ private VirtualMerkleStateInitializer() {}
...
-                        .storageDir(pathToJasperDBStorageDir)
-                        .maxNumOfKeys(maximumNumberOfKeyValuePairsCreation)
-                        .hashesRamToDiskThreshold(0)
-                        .preferDiskBasedIndexes(false);
-
+                        maximumNumberOfKeyValuePairs);
                 final VirtualMap<SmartContractMapKey, SmartContractMapValue> virtualMap =
-                        new VirtualMap<>("smartContracts", jasperDbBuilder);
+                        createSmartContractsVM(useMerkleDb, storageDir, maximumNumberOfKeyValuePairs);
+                logger.info(LOGM_DEMO_INFO, "SC VM = {}, DS = {}", virtualMap, virtualMap.getDataSource());
```
Comment (jsync-swirlds) [inline platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/virtualmerkle/state/VirtualMerkleStateInitializer.java:112]: Quite a few of these info logs probably should be debug.  Info is actually a fairly high logging level.

**Review by jsync-swirlds — APPROVED** [review]
Some nitpicks about logging level, otherwise looks good.

**Review by jsync-swirlds — APPROVED** [review]

**Review by imalygin — APPROVED** [review]

**Review by imalygin — APPROVED** [review]

**Review by nathanklick — APPROVED** [review]

**Review by timo0 — APPROVED** [review]

**Review by iwsimon — APPROVED** [review]

**Review by poulok — APPROVED** [review]

**Review by jasperpotts — COMMENTED** [review]
Looks reasonable

**Pull request merged**